modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/ColocationGroup.java

```diff
@@ -47,12 +49,7 @@
 /** */
 public class ColocationGroup implements MarshalableMessage {
     /** */
-    private static final int SYNTHETIC_PARTITIONS_COUNT =
-        IgniteSystemProperties.getInteger("IGNITE_CALCITE_SYNTHETIC_PARTITIONS_COUNT", 512);
-
-    /** */
-    @GridDirectCollection(Long.class)
-    private List<Long> sourceIds;
+    private long[] sourceIds;
```

Comment:
GridIntList may be ?

Reply:
What for? The regular array covers all needs and it's faster to send an array than an array wrapped into `Message`.